src/Http/Routing/test/UnitTests/Builder/RequestDelegateEndpointRouteBuilderExtensionsTest.cs

```diff
@@ -69,6 +69,27 @@ public void MapEndpoint_StringPattern_BuildsEndpoint()
         Assert.Equal("/", endpointBuilder1.RoutePattern.RawText);
     }
 
+    [Fact]
+    public void MapEndpoint_ReturnGenericTypeTask_GeneratedDelegate()
```

Comment:
Add an assert that observes we write to the body.

Reply:
how to assert body in unit test?

Reply:
Create a mock `HttpContext` with a `MemoryStream` as the `HttpResponse.Body`. Then call `endpointBuilder1.RequestDelegate` with the mock. Once that's done, verify that "response" was written to the MemoryStream. Here's a similar example:

https://github.com/dotnet/aspnetcore/blob/95dcd9d252842cabe4a25fb07a67c0c220ae9807/src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs#L2501-L2518